capplets/display/xrandr-capplet.c

```diff
@@ -452,6 +457,54 @@ mirror_screens_is_supported (App *app)
     return mirror_is_supported;
 }
 
+static void
+on_scale_btn_active_changed_cb (GtkWidget *widget,
+                                App       *app)
+{
+    gint scale;
+
+    if (!gtk_toggle_button_get_active (GTK_TOGGLE_BUTTON (widget)))
+        return;
+
+    scale = GPOINTER_TO_INT (g_object_get_data (G_OBJECT (widget), "scale"));
+    g_settings_set_int (app->scale_settings, WINDOW_SCALE_KEY, scale);
+}
+
+static void
+rebuild_scale_window (App *app)
+{
+    GtkRadioButton *group = NULL;
+    gint32          scale;
+    int             i;
+    const char     *button_label[] = {_("auto detect"), _("100%"), _("200%"), NULL};
+
+    /* destroy any previous scale_btn */
+    gtk_container_foreach (GTK_CONTAINER (app->scale_bbox), G_CALLBACK (gtk_widget_destroy), NULL);
+    gtk_container_foreach (GTK_CONTAINER (app->scale_bbox), (GtkCallback) foreach_destroy_cb, NULL);
```

Comment:
foreach_destroy_cb doen't exist now.

Reply:
You deleted the wrong line.

Reply:
It's my fault. I'm in a hurry to go home for dinner